### PR TITLE
Feature/search by tag

### PR DESCRIPTION
**Pull request by knees4bees**
--->  ((are you in the right branch??))
## What does this PR do?
- Creates filter-by-tag capability and combines it with search so that users can search among a subset of recipes (e.g., breakfast recipes that include chocolate).
- Adds tags to the list of things that are searched for using user's search bar input (previously just ingredient and recipe name).
- Removes separate search button (now just one "go" button).
- Adds a dynamic title to recipe list view to describe the list of recipes (e.g. "Breakfast recipes matching "chocolate" ").

## How should it be tested?
 - 1. Double-check that all tests are still passing (they were at the time of writing).
 - 2. In the UI, try selecting all categories from the dropdown, with and without something typed in the search bar, and confirm that the results make sense. Also try searching for a tag like "snack" or "birthday" that isn't included in the dropdown categories and make sure results appear.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -77,6 +63,8 @@ const displayPantry = () => {
 const displayRandomFavorites = () => {
   let chunk = '';
   let fourRandomRecipes = [];
+
+  // TODO need to use one of the new prototype methods for this?
```
Comment (piknikki) [inline src/scripts.js:67]: I only want to run it 4 times, so it's a good use case for a for loop, as opposed to an iterator method.

Comment (piknikki) [reply src/scripts.js:67]: But I'm open to suggestions on iterator methods that could get only four random recipes.

File: src/scripts.js [hunk 2]
```diff
@@ -199,122 +184,130 @@ const displayRecipe = (id) => {
...
+    // TODO also starter, hor d'oeuvre, hor d'oevres, antipasti, antipasto
+  } else if (choice === 'side-dishes') {
+    searchFor = 'side dish';
+  } else if (choice === 'main-courses') {
+    searchFor = 'main dish';
+    // TODO also main course, lunch, dinner
+  } else if (choice === 'desserts') {
+    searchFor = 'dessert';
+  } else {
+    searchFor = choice;
```
Comment (piknikki) [inline src/scripts.js:240]: Maybe we can refactor this to use a switch!!

**Review by piknikki — COMMENTED** [review]
Great job with this. Let's talk about my notes during our standup today.

File: src/scripts.js (hunk 1 above)
Comment (knees4bees) [reply src/scripts.js:67]: On our list to ask David about

File: src/scripts.js (hunk 2 above)
Comment (knees4bees) [reply src/scripts.js:240]: YES

**Comment by m1073496** [comment]
Looks great! Conflicts are resolved (-:

**Pull request merged**